packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -3,34 +3,27 @@
  */
 // eslint-disable-next-line no-restricted-imports
 import type * as Ariakit from '@ariakit/react';
+import type { FocusEventHandler, MouseEventHandler } from 'react';
 
-export type CustomSelectContext =
-	| {
-			/**
-			 * The store object returned by Ariakit's `useSelectStore` hook.
-			 */
-			store: Ariakit.SelectStore;
-	  }
-	| undefined;
-
-export type CustomSelectProps = {
+export type CustomSelectStore = {
 	/**
-	 * The child elements. This should be composed of CustomSelectItem components.
+	 * The store object returned by Ariakit's `useSelectStore` hook.
 	 */
-	children: React.ReactNode;
+	store: Ariakit.SelectStore;
+};
+
+export type CustomSelectContext = CustomSelectStore | undefined;
+
+export type CustomSelectButtonProps = {
 	/**
 	 * An optional default value for the control. If left `undefined`, the first
 	 * non-disabled item will be used.
 	 */
 	defaultValue?: string | string[];
-	/**
-	 * Label for the control.
-	 */
-	label: string;
 	/**
 	 * A function that receives the new value of the input.
 	 */
-	onChange?: ( newValue: string | string[] ) => void;
+	onSelectionChange?: ( newValue: string | string[] ) => void;
```

Comment:
And now I'm thinking it would be better to omit it and keep the naming which I've done here: https://github.com/WordPress/gutenberg/pull/57902/commits/940cdb1eed59d625ec97d2c02d89fe5d8645cd86